src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -128,7 +130,8 @@ public NodeEnvironment(Settings settings, Environment environment) throws IOExce
         int maxLocalStorageNodes = settings.getAsInt("node.max_local_storage_nodes", 50);
         for (int possibleLockId = 0; possibleLockId < maxLocalStorageNodes; possibleLockId++) {
             for (int dirIndex = 0; dirIndex < environment.dataWithClusterFiles().length; dirIndex++) {
-                Path dir = environment.dataWithClusterFiles()[dirIndex].resolve(Paths.get(NODES_FOLDER, Integer.toString(possibleLockId)));
+                // TODO: wtf with resolve(get())
+                Path dir = environment.dataWithClusterFiles()[dirIndex].resolve(PathUtils.get(NODES_FOLDER, Integer.toString(possibleLockId)));
```

Comment:
lets do it as a followup, we should try to minimize non-test changes here, it was just a TODO since this was using Paths.get for "concatenation" (which is not recommended)